### ISSUE TITLE
Expose Jetty task queue configuration in ring-jetty-adapter

### ISSUE THREAD

**Issue by andrewmcveigh** [ring-clojure/ring#272]
We need to be able to configure the length of Jetty's task queue so that we can drop work that our app can't handle[1].

We've had to work this code into our application for the moment. It "works", but it would be much nicer to get this option into ring-jetty-adapter. It would mean exposing in the ring-jetty-adapter API at least the thread `idleTimeout` (or defaulting) and `queue` from the QueuedThreadPool[2].

I can offer a pull-request if this is something you'd consider.

[1] https://wiki.eclipse.org/Jetty/Howto/High_Load#Thread_Pool
[2] http://download.eclipse.org/jetty/9.3.12.v20160915/apidocs/org/eclipse/jetty/util/thread/QueuedThreadPool.html

**Comment by weavejester** [comment]
Can you give some more background and code on this? Presumably you're using the Jetty `:configurator` option?

**Comment by andrewmcveigh** [comment]
Sure.

We're having to create the server with a bit of modified code to create the `ThreadPool`.

The reason we aren't using the `:configurator` option is that it only gives us access to the already created server. There's no way to `setThreadPool`, and the `QueuedThreadPool` needs to be constructed with a task queue if you want any control over it. (OK, there's a deprecated method `setQueue`[1] on the `QueuedThreadPool` but we'd rather not rely on that.)

The code below is not the exact code we're using atm, but I think it shows the intent.
```clojure
(defn- create-threadpool [options]
  (let [task-queue-length (options :task-queue-length)
        pool (if task-queue-length
               (QueuedThreadPool. (options :max-threads 50)
                                  (options :min-threads 8)
                                  (options :thread-idle-time 60000)
                                  (ArrayBlockingQueue. task-queue-length))
               (QueuedThreadPool. (options :max-threads 50)
                                  (options :min-threads 8)
                                  (options :thread-idle-time 60000)))]
    (when (:daemon? options false)
      (.setDaemon pool true))
    pool))
```

[1] http://download.eclipse.org/jetty/9.3.12.v20160915/apidocs/org/eclipse/jetty/util/thread/QueuedThreadPool.html#setQueue-java.util.concurrent.BlockingQueue-

**Comment by matrix10657** [comment]
Looks like this has been merged into master, and this would be a great help to me, personally.

Any chance we could close this out and get a 1.6.3 release?

**Comment by weavejester** [comment]
I can close it, but as it's a feature it will need to wait for Ring 1.7. I'll check to see whether everything scheduled for that release is done.

**Comment by weavejester** [comment]
Fixed by #289.
